tests/network/vmi_istio.go

```diff
@@ -180,42 +220,70 @@ var _ = SIGDescribe("[Serial] Istio", func() {
+				})
+			})
+			Context("VMI with no explicit ports", func() {
+				BeforeEach(func() {
+					vmiPorts = []v1.Port{}
+				})
+				It("Should be able to reach http server outside of mesh", func() {
+					Expect(
+						checkHTTPServiceReturnCode(serverVMIAddress, testPort, generateExpectedHTTPReturnCodeRegex("200")),
+					).ToNot(HaveOccurred())
+				})
 			})
+
 			Context("With Sidecar allowing only registered external services", func() {
+				// Istio Envoy will intercept the request because of the OutboundTrafficPolicy set to REGISTRY_ONLY.
+				// Envoy responds with 502 Bad Gateway return code.
+				// After Sidecar with OutboundTrafficPolicy is created, it may take a while for the Envoy proxy
+				// to sync with the change, first request may still get through, hence the Eventually used for assertions.
+
 				BeforeEach(func() {
 					sidecarRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "sidecars"}
 					registryOnlySidecar := generateRegistryOnlySidecar()
 					_, err = virtClient.DynamicClient().Resource(sidecarRes).Namespace(tests.NamespaceTestDefault).Create(context.TODO(), registryOnlySidecar, metav1.CreateOptions{})
 					Expect(err).ShouldNot(HaveOccurred())
 				})
 
-				// Istio Envoy will intercept the request because of the OutboundTrafficPolicy set to REGISTRY_ONLY.
-				// Envoy responds with 502 Bad Gateway return code.
-				// After Sidecar with OutboundTrafficPolicy is created, it may take a while for the Envoy proxy
-				// to sync with the change, first request may still get through, hence the Eventually.
-				It("Should not be able to reach http server outside of mesh", func() {
-					Eventually(func() error {
-						return console.SafeExpectBatch(vmi, []expect.Batcher{
-							&expect.BSnd{S: "\n"},
-							&expect.BExp{R: console.PromptExpression},
-							&expect.BSnd{S: curlCommand(serverVMIAddress, testPort)},
-							&expect.BExp{R: generateExpectedHTTPReturnCodeRegex("5..")},
-							&expect.BSnd{S: "echo $?\n"},
-							&expect.BExp{R: console.RetValue("0")},
-						}, 60)
-					}, 90*time.Second, 10*time.Second)
+				Context("VMI with explicit ports", func() {
+					BeforeEach(func() {
+						vmiPorts = explicitPorts
+					})
+					It("Should not be able to reach http service outside of mesh", func() {
+						Eventually(func() error {
+							return checkHTTPServiceReturnCode(serverVMIAddress, testPort, generateExpectedHTTPReturnCodeRegex("5.."))
+						}, externalServiceCheckTimeout, externalServiceCheckInterval)
+					})
+				})
+				Context("VMI with no explicit ports", func() {
+					BeforeEach(func() {
+						vmiPorts = []v1.Port{}
+					})
+					It("Should not be able to reach http service outside of mesh", func() {
+						Eventually(func() error {
+							return checkHTTPServiceReturnCode(serverVMIAddress, testPort, generateExpectedHTTPReturnCodeRegex("5.."))
+						}, externalServiceCheckTimeout, externalServiceCheckInterval)
+					})
```

Comment:
Please add tests testing the pre-defined ports are not forwarded to the vm.

Reply:
Added [this test](https://github.com/kubevirt/kubevirt/pull/5461/commits/6ad006b36a9513a1eeefbfd5983f45b448dec462#diff-549b5c85962b6a2029257ee04a44302196d99e8c66fe90d9b5032915f3731344R159)
I presume testing one of the restricted porst should be enough, since we treat them all the same, right?